public/app/plugins/datasource/elasticsearch/configuration/DataLink.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { css } from 'emotion';
+import { Button, FormField, VariableSuggestion, DataLinkInput, stylesFactory } from '@grafana/ui';
+import { DataLinkConfig } from '../types';
+
+const getStyles = stylesFactory(() => ({
+  firstRow: css`
+    display: flex;
+  `,
+  nameField: css`
+    flex: 2;
+  `,
+  regexField: css`
+    flex: 3;
+  `,
+}));
+
+type Props = {
+  value: DataLinkConfig;
+  onChange: (value: DataLinkConfig) => void;
+  onDelete: () => void;
+  suggestions: VariableSuggestion[];
+  className?: string;
+};
+export const DataLink = (props: Props) => {
+  const { value, onChange, onDelete, suggestions, className } = props;
+  const styles = getStyles();
+
+  const handleChange = (field: keyof typeof value) => (event: React.ChangeEvent<HTMLInputElement>) => {
+    onChange({
+      ...value,
+      [field]: event.currentTarget.value,
+    });
+  };
+
+  return (
+    <div className={className}>
+      <div className={styles.firstRow}>
+        <FormField
+          className={styles.nameField}
+          labelWidth={5}
+          // A bit of a hack to prevent using default value for the width from FormField
+          inputWidth={null}
+          label="Pattern"
```

Comment:
Should be "Field"?

Reply:
Not sure what to call it, it can be a regex pattern actually that can match multiple fields or just multiple variation of the same field (like `/TraceI(d|D)/`) so seems like calling it a `Field` people could miss that.

Reply:
Hmm, I think the use case will overwhelmingly be field name. I'd rather let people find out or document that regexps are supported.